lib/hooks/core/hook-http.js

```diff
@@ -51,13 +51,13 @@ function requestWrap(request) {
 
     var namespace = cls.getNamespace();
     var uri = uriFromOptions(options);
-    var currContext = cls.getRootContext().createChildSpanData(uri);
-    currContext.span.setLabel(TraceLabels.HTTP_METHOD_LABEL_KEY, options.method);
-    currContext.span.setLabel(TraceLabels.HTTP_URL_LABEL_KEY, uri);
+    var span = agent.startSpan(uri);
+    span.addLabel(TraceLabels.HTTP_METHOD_LABEL_KEY, options.method);
```

Comment:
You mean remove it from the `OpaqueSpan` interface? It should be removed, or at the least, made private.

I think I might have not done a good job explaining the motivation behind why we want to do this. We want to use the public agent API instead of the private API for a few reasons:
* DRY. There should be only one way of doing something.
* If the public API is sufficient for a use-case, it should be used instead of a private API. This reduces tight coupling between internal components.
* It gets the public API tested for usefulness and finds the deficiencies. If there is use-cases that it doesn't solve for us, it is probably going to let external users down as well.

Hope this makes sense.